zebedee-reader/src/test/java/com/github/onsdigital/zebedee/reader/api/bean/PublishedIndexResponseTest.java

```diff
@@ -0,0 +1,84 @@
+    }
+
+    @Test
+    public void testAddDocument_noDocuments() {
+        PublishedIndexResponse publishedIndexResponse = new PublishedIndexResponse();
+        publishedIndexResponse.addDocuments(listOfDocuments());
+        assertEquals(publishedIndexResponse.getCount(),0);
+        assertNotNull(publishedIndexResponse.getItems());
+        assertEquals(publishedIndexResponse.getItems().size(),0);
+    }
+
+    @Test
+    public void testAddDocument_oneDocument() {
+        PublishedIndexResponse publishedIndexResponse = new PublishedIndexResponse();
+        publishedIndexResponse.addDocuments(listOfDocuments(URI1));
+        assertEquals(publishedIndexResponse.getCount(),1);
+        assertNotNull(publishedIndexResponse.getItems());
+        assertEquals(publishedIndexResponse.getItems().size(),1);
+        assertNotNull(publishedIndexResponse.getItems().get(0));
+        assertEquals(publishedIndexResponse.getItems().get(0).getUri(),URI1);
+    }
+
+    @Test
+    public void testAddDocument_twoDocuments() {
+        PublishedIndexResponse publishedIndexResponse = new PublishedIndexResponse();
+        publishedIndexResponse.addDocuments(listOfDocuments(URI1,URI2));
+        assertEquals(publishedIndexResponse.getCount(),2);
+        assertNotNull(publishedIndexResponse.getItems());
+        assertEquals(publishedIndexResponse.getItems().size(),2);
+        assertNotNull(publishedIndexResponse.getItems().get(0));
+        assertEquals(publishedIndexResponse.getItems().get(0).getUri(),URI1);
+        assertNotNull(publishedIndexResponse.getItems().get(1));
+        assertEquals(publishedIndexResponse.getItems().get(1).getUri(),URI2);
+    }
+
+    @Test
+    public void testAddDocument_multipleCalls() {
+        PublishedIndexResponse publishedIndexResponse = new PublishedIndexResponse();
+        publishedIndexResponse.addDocuments(listOfDocuments(URI1));
+        assertEquals(publishedIndexResponse.getCount(),1);
+        assertNotNull(publishedIndexResponse.getItems());
+        assertEquals(publishedIndexResponse.getItems().size(),1);
+        assertNotNull(publishedIndexResponse.getItems().get(0));
+        assertEquals(publishedIndexResponse.getItems().get(0).getUri(),URI1);
+        
+        publishedIndexResponse.addDocuments(listOfDocuments(URI2));
+        assertEquals(publishedIndexResponse.getCount(),2);
+        assertNotNull(publishedIndexResponse.getItems());
+        assertEquals(publishedIndexResponse.getItems().size(),2);
+        assertNotNull(publishedIndexResponse.getItems().get(0));
+        assertEquals(publishedIndexResponse.getItems().get(0).getUri(),URI1);
+        assertNotNull(publishedIndexResponse.getItems().get(1));
+        assertEquals(publishedIndexResponse.getItems().get(1).getUri(),URI2);
+    }
+
+    private List<Document> listOfDocuments(String... uris) {
+        return Arrays.stream(uris)
+                .map(u -> new Document(u,null))
+                .collect(Collectors.toList());
+    }
+}
```

Comment:
* no tests for the ability to set/get limit, offset or totalCount
* no test for serialized name of totalCount

Reply:
was able to get tests passing with the following implementation:
```
package com.github.onsdigital.zebedee.reader.api.bean;

import com.github.onsdigital.zebedee.search.indexing.Document;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * Object representing a response for the PublishedIndex endpoint
 */
public class PublishedIndexResponse {

    private int count;
    private List<Item> items;
//    private int limit;
//    private int offset;
//    @SerializedName("total_count")
//    private int totalCount;

    public PublishedIndexResponse() {
        this.items = new ArrayList<>();
    }

    /**
     * Adds a list of documents to the response as Items. Also updates the
     *
     * @param docs  List of Documents to add to response
     */
    public void addDocuments(List<Document> docs) {
        if (docs == null) {
            return;
        }
        docs.stream()
                .map(d -> new Item(d.getUri()))
                .forEach(d -> this.items.add(d));

        count += docs.size();
    }

    public int getCount() {
        return count;
    }

    public List<Item> getItems() {
        return items;
    }
//
//    public int getLimit() {
//        return limit;
//    }
//
    public void setLimit(int limit) {
        //this.limit = limit;
    }
//
//    public int getOffset() {
//        return offset;
//    }
//
    public void setOffset(int offset) {
        //this.offset = offset;
    }
//
//    public int getTotalCount() {
//        return totalCount;
//    }
//
    public void setTotalCount(int totalCount) {
//        this.totalCount = totalCount;
    }
//
    /**
     * Sub-object representing an individual document returned by the PublishedIndex
     */
    public class Item {
        private String uri;

        public Item(String uri) {
            this.uri = uri;
        }

        public String getUri() {
            return uri;
        }
    }
}
```

Reply:
We don't test getters and setters. But I've added a test for the serialisation.